iceoryx_posh/source/runtime/service_discovery.cpp

```diff
@@ -29,17 +29,12 @@ ServiceContainer ServiceDiscovery::findService(const cxx::optional<capro::IdStri
         m_serviceRegistry = *serviceRegistrySample;
     });
 
-    roudi::ServiceRegistry::ServiceDescriptionVector_t tempSearchResult;
-    m_serviceRegistry.find(tempSearchResult, service, instance, event);
-
     ServiceContainer searchResult;
```

Comment:
These containers on the stack are a problem, currently with MacOS (CI fixed). However, the best way to solve this is to avoid them in intermediate computation if possible.

If the final result is such a container, we have no choice  other than changing the API. Basically there are two immediate options: 

1) The callback/filter API which #1105 introduces or, 
2) as a specialized version, allowing the container to be filled passed in by the like `template<typename Container> find( ..., Container&);`

Now the user can choose the container, but if she chooses poorly it will overflow and this requires well-defined handling.

1. is most generic and shifts the responsibility completely to the user. However, it may be not intuitive for some.
2. is more intuitive and the user has the choice of container.

I do not propose to introduce 2. now (requires testing etc.), but in the long run we need to deal with those problems.